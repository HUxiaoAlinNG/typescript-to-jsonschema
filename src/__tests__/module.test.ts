import * as path from 'path';

import genTypeSchema from '../index';

const file = path.join(__dirname, `../../jest/module.ts`);

genTypeSchema.genJsonDataFormFile(file);

const getSchema = (type: string) => {
  const schema = genTypeSchema.getJsonSchema(file, type);
  return schema;
};

test('Module重命名模块_0', () => {
  expect(getSchema('Module_0')).toMatchSnapshot();
});

test('Module重命名模块多文件_0', () => {
  expect(getSchema('Module_1')).toMatchSnapshot();
});

test('Module多文件全部导出_0', () => {
  expect(getSchema('Module_2')).toMatchSnapshot();
});

test('Module多文件按组导出_0', () => {
  expect(getSchema('Module_3')).toMatchSnapshot();
});

test('Module多文件按组默认导出_0', () => {
  expect(getSchema('Module_4')).toMatchSnapshot();
});

test('Module多文件命名空间导出_0', () => {
  expect(getSchema('Module_5')).toMatchSnapshot();
});

test('Module多文件默认导出_0', () => {
  expect(getSchema('Module_6')).toMatchSnapshot();
});
