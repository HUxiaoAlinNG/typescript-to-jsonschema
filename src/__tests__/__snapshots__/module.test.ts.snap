// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module多文件全部导出_0 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "ExportAll": Object {
      "additionalProperties": false,
      "properties": Object {
        "count": Object {
          "type": "number",
        },
      },
      "required": Array [
        "count",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "attr": Object {
      "$ref": "#/definitions/ExportAll",
    },
  },
  "required": Array [
    "attr",
  ],
  "type": "object",
}
`;

exports[`Module多文件命名空间导出_0 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "nameSpaceExport.User": Object {
      "additionalProperties": false,
      "properties": Object {
        "name": Object {
          "type": "string",
        },
      },
      "required": Array [
        "name",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "attr": Object {
      "$ref": "#/definitions/nameSpaceExport.User",
    },
  },
  "required": Array [
    "attr",
  ],
  "type": "object",
}
`;

exports[`Module多文件按组导出_0 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "GroupExport": Object {
      "additionalProperties": false,
      "properties": Object {
        "attr": Object {
          "type": "string",
        },
      },
      "required": Array [
        "attr",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "attr": Object {
      "$ref": "#/definitions/GroupExport",
    },
  },
  "required": Array [
    "attr",
  ],
  "type": "object",
}
`;

exports[`Module多文件按组默认导出_0 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "GroupDefaultExport": Object {
      "additionalProperties": false,
      "properties": Object {
        "name": Object {
          "type": "string",
        },
      },
      "required": Array [
        "name",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "attr": Object {
      "$ref": "#/definitions/GroupDefaultExport",
    },
  },
  "required": Array [
    "attr",
  ],
  "type": "object",
}
`;

exports[`Module多文件默认导出_0 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "DefaultExport": Object {
      "additionalProperties": false,
      "properties": Object {
        "name": Object {
          "type": "string",
        },
      },
      "required": Array [
        "name",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "attr": Object {
      "$ref": "#/definitions/DefaultExport",
    },
  },
  "required": Array [
    "attr",
  ],
  "type": "object",
}
`;

exports[`Module重命名模块_0 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "_AAA": Object {
      "additionalProperties": false,
      "properties": Object {
        "other1": Object {
          "type": "string",
        },
      },
      "required": Array [
        "other1",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "attr": Object {
      "$ref": "#/definitions/_AAA",
    },
  },
  "required": Array [
    "attr",
  ],
  "type": "object",
}
`;

exports[`Module重命名模块多文件_0 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {
    "_AAA": Object {
      "additionalProperties": false,
      "properties": Object {
        "other1": Object {
          "type": "string",
        },
      },
      "required": Array [
        "other1",
      ],
      "type": "object",
    },
    "_User": Object {
      "additionalProperties": false,
      "properties": Object {
        "name": Object {
          "type": "string",
        },
      },
      "required": Array [
        "name",
      ],
      "type": "object",
    },
  },
  "properties": Object {
    "attr1": Object {
      "$ref": "#/definitions/_AAA",
    },
    "attr2": Object {
      "$ref": "#/definitions/_User",
    },
  },
  "required": Array [
    "attr1",
    "attr2",
  ],
  "type": "object",
}
`;
